### PR TITLE
Loadout mods tooltips

### LINKED ISSUE #8973: Mod placement should have rich mod tooltips
Right now it has just title tooltips

### PR DESCRIPTION
**Pull request by bhollis**
We'd lost our tooltips (or never had them) for mods in a lot of places especially in the Loadout Optimizer and Mod Assignment pages. This brings them back, and also generally cleans up the structure of a lot of things. And I renamed CompareDrawer to CompareLoadoutDrawer because I always open it thinking it's the Compare drawer.

Fixes #8973

### CODE REVIEW

File: src/app/inventory/InventoryItem.m.scss [hunk 1]
```diff
@@ -16,10 +16,6 @@
   height: var(--item-size);
   // searchHidden will adjust opacity/transform, this transitions them
   transition: opacity 0.2s, transform 0.2s;
-
-  > * {
-    pointer-events: none;
```
Comment (bhollis) [inline src/app/inventory/InventoryItem.m.scss:21]: This was the culprit for no tooltips.

File: src/app/loadout-builder/generated-sets/CompareLoadoutsDrawer.m.scss [hunk 2]
```diff
@@ -7,7 +7,6 @@
 
 .content {
   margin: 16px;
-  display: inline-block;
```
Comment (bhollis) [inline src/app/loadout-builder/generated-sets/CompareLoadoutsDrawer.m.scss:10]: This made loadouts in the compare-loadouts drawer oddly squished

**Review by vivekhnz — APPROVED** [review]
Nice 👍!

One minor thing I noticed is that intrinsic plug icons are _slightly_ smaller after applying the `borderless` class to them.

dim-intrinsic-plug-icon-size

This is because the item icons use `border-box` and the `borderless` class doesn't zero out the `border-width`.

https://github.com/DestinyItemManager/DIM/blob/0e87685bb84f8aeb35e25bdc5824f5beb1e39c6a/src/app/inventory/ItemIcon.m.scss#L60-L62

`border: none` seems to fix the issue for intrinsic plugs and still looks fine for other borderless items e.g. vendor engrams.

**Review by nev-r — APPROVED** [review]
nicer thank you

**Review by robojumper — APPROVED** [review]
I've learned to not get confused by the `CompareDrawer`'s name but this is definitely clearer.

**Comment by bhollis** [comment]
The idea behind the `borderless` class still contributing a border is because item icons are 96px square, and our default item size is 50px. So you remove the 2px of border and get 48px, which is exactly half the original image size for perfect high DPI screen scaling. It doesn't quite work that way on the scaled down icons in the Loadout Optimizer but in general that's the idea.

**Pull request merged**